### ISSUE TITLE
W1404:  implicit-str-concat is not working with f strings  

### ISSUE THREAD

**Issue by klaucos** [pylint-dev/pylint#8968]
### Current problem

W1404 doesn't work for implicit concatenation of f strings.

Example: 
`raise ValueError(f"{a} = " f"{b}.")`

It correctly raises error for: 
`raise ValueError("a =" "b.")`

### Desired solution

`raise ValueError(f"{a} =" f"{b}.") `
should raise 
`Implicit string concatenation found in %s`

### Additional context

These weird f string breaks can be generated when migrating to Black formater.

**Comment by Timmmm** [comment]
I think it's [this](https://github.com/pylint-dev/pylint/blob/866e8ad19d12a06234efe90cd75b292c4956e052/pylint/checkers/strings.py#L876C1-L878) condition that needs to change.

```
            if not (
                isinstance(elt, nodes.Const) and elt.pytype() in _AST_NODE_STR_TYPES
            ):
                continue
```

Unfortunately this is likely to be very difficult because Pylint uses the Astroid Python parser and it completely discards any information about concatenation of f-strings.

```
>>> print(astroid.extract_node('f"foo {1}" "bar"').repr_tree())
JoinedStr(values=[
      Const(
         value='foo ',
         kind=None),
      FormattedValue(
         conversion=-1,
         value=Const(
            value=1,
            kind=None),
         format_spec=None),
      Const(
         value='bar',
         kind=None)])
>>> print(astroid.extract_node('f"foo {1}bar"').repr_tree())
JoinedStr(values=[
      Const(
         value='foo ',
         kind=None),
      FormattedValue(
         conversion=-1,
         value=Const(
            value=1,
            kind=None),
         format_spec=None),
      Const(
         value='bar',
         kind=None)])
```
